lib/storage/trustedcluster.go

```diff
@@ -323,13 +323,19 @@ func (c *TrustedClusterV2) CheckAndSetDefaults() error {
 	if c.Spec.ReverseTunnelAddress == "" {
 		return trace.BadParameter("tunnel_addr can't be empty")
 	}
+	if len(c.Spec.Roles) != 0 && len(c.Spec.RoleMap) != 0 {
+		return trace.BadParameter("either roles or role_map should be set, not both")
```

Comment:
Would it hurt if this reset the roles when a role map has been specified or there's no implicit precedence to decide which to use over which?

Reply:
Yeah, I would not make assumptions for the user to avoid surprises.